src/Php/PhpVersionResolver/ProjectComposerJsonPhpVersionResolver.php

```diff
@@ -28,21 +28,21 @@ public static function resolve(string $composerJson): ?int
         $composerJsonContents = FileSystem::read($composerJson);
         $projectComposerJson = Json::decode($composerJsonContents, Json::FORCE_ARRAY);
 
+        // give this one a priority, as more generic one
+        $requirePhpVersion = $projectComposerJson['require']['php'] ?? null;
+        if ($requirePhpVersion !== null) {
+            self::$cachedPhpVersions[$composerJson] = self::createIntVersionFromComposerVersion($requirePhpVersion);
+            return self::$cachedPhpVersions[$composerJson];
+        }
+
         // see https://getcomposer.org/doc/06-config.md#platform
         $platformPhp = $projectComposerJson['config']['platform']['php'] ?? null;
         if ($platformPhp !== null) {
             self::$cachedPhpVersions[$composerJson] = PhpVersionFactory::createIntVersion($platformPhp);
             return self::$cachedPhpVersions[$composerJson];
         }
 
-        $requirePhpVersion = $projectComposerJson['require']['php'] ?? null;
-        if ($requirePhpVersion === null) {
-            return self::$cachedPhpVersions[$composerJson] = null;
-        }
-
-        self::$cachedPhpVersions[$composerJson] = self::createIntVersionFromComposerVersion($requirePhpVersion);
-
-        return self::$cachedPhpVersions[$composerJson];
+        return null;
```

Comment:
null result still need to be cached:

```suggestion
        return self::$cachedPhpVersions[$composerJson] = null;
```

I will create new PR for it.

Reply:
Thanks

Reply:
https://github.com/rectorphp/rector-src/pull/5559